studio/components/layouts/SQLEditorLayout/SQLEditorLayout.tsx

```diff
@@ -1,37 +1,55 @@
 import { FC, ReactNode, useEffect } from 'react'
 import { observer } from 'mobx-react-lite'
 
-import { useStore } from 'hooks'
+import { useStore, withAuth } from 'hooks'
 import Error from 'components/ui/Error'
+import { createSqlEditorStore, SqlEditorContext } from 'localStores/sqlEditor/SqlEditorStore'
 import ProjectLayout from '../ProjectLayout/ProjectLayout'
 import SQLEditorMenu from './SQLEditorMenu'
+import { useState } from 'react'
 
 interface Props {
   title: string
   children: ReactNode
 }
 
 const SQLEditorLayout: FC<Props> = ({ title, children }) => {
-  const { content } = useStore()
+  const { ui, content, meta } = useStore()
+  const { profile: user } = ui
 
+  const [sqlEditorStore, setSqlEditorStore] = useState<any | null>(null)
```

Comment:
quick question - why are we keeping the store in a useState? shouldn't the store be treated as sort of a global state that can be accessed via a `useXXX` hook? that's the practice that we've been following at least

wanna clarify cause i'd ideally want to consume stores consistently across the dashboard. if this is a better practice then perhaps we need to doc it somewhere so that we can refactor for the other pages too

Reply:
Good point - I did it this way to always ensure the `sqlEditorStore` had an up-to-date meta store and `projectRef`; however, this might lead to unnecessary loading in the UI. I'll do some more testing and get back to you!

Reply:
just to check in on this if there's any follow up? so we don't block your stuff from going in

if its sound to just access the sqlEditorStore via `useSqlEditorStore` (and remove `createSqlEditorStore`) to keep things consistent for now think that'll be good

Reply:
I've switched to using `useSqlEditorStore` and not state :)